### PR TITLE
Adding Support for Jira Integration

### PR DESCRIPTION
**Pull request by bala-ceg**
**Jira New Integration Support**
Added support as per the [Integration Guide](https://docs.mindsdb.com/contribute/integrations).
Please let me know if this is fine and help me resolve this error

**Testing via MindsDB UI**

Screenshot 2022-11-27 at 10 02 14 PM
Screenshot 2022-11-27 at 10 03 25 PM

**Unit Test Results**

Screenshot 2022-11-27 at 10 26 18 PM

**Prediction Output**
Screenshot 2022-11-27 at 10 58 05 PM

Screenshot 2022-11-27 at 11 18 28 PM

**Error Faced** 

```
INFO:lightwood-33203:No torchvision detected, image helpers not supported.
INFO:lightwood-33203:No torchvision/pillow detected, image encoder not supported
/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb-venv/lib/python3.8/site-packages/gluonts/json.py:101: UserWarning: Using `json`-module for json-handling. Consider installing one of `orjson`, `ujson` to speed up serialization and deserialization.
  warnings.warn(
/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb-venv/lib/python3.8/site-packages/gluonts/model/deepar/__init__.py:18: FutureWarning: The module gluonts.model.deepar has been moved to gluonts.mx.model.deepar. In GluonTS v0.12 it will be no longer possible to use the old path. Try to use 'from gluonts.mx import DeepAREstimator'.
  warnings.warn(
INFO:type_infer-33203:Analyzing a sample of 50
INFO:type_infer-33203:from a total population of 50, this is equivalent to 100.0% of your data.
INFO:type_infer-33203:Infering type for: Ticket_No
INFO:type_infer-33203:Column Ticket_No has data type quantity
INFO:type_infer-33203:Infering type for: Summary
INFO:type_infer-33203:Doing text detection for column: Summary
INFO:type_infer-33203:Infering type for: Status
INFO:type_infer-33203:Column Status has data type categorical
INFO:type_infer-33203:Infering type for: Reporter
INFO:type_infer-33203:Column Reporter has data type categorical
INFO:type_infer-33203:Infering type for: Assignee
INFO:type_infer-33203:Doing text detection for column: Assignee
INFO:type_infer-33203:Column Assignee has data type categorical
INFO:type_infer-33203:Infering type for: Priority
INFO:type_infer-33203:Column Priority has data type binary
INFO:lightwood-33203:Starting statistical analysis
INFO:lightwood-33203:Finished statistical analysis
INFO:lightwood-33203:Unable to import black formatter, predictor code might be a bit ugly.
INFO:lightwood-33203:[Learn phase 1/8] - Statistical analysis
INFO:lightwood-33203:Starting statistical analysis
INFO:lightwood-33203:Finished statistical analysis
DEBUG:lightwood-33203: `analyze_data` runtime: 0.02 seconds
INFO:lightwood-33203:[Learn phase 2/8] - Data preprocessing
INFO:lightwood-33203:Cleaning the data
DEBUG:lightwood-33203: `preprocess` runtime: 0.0 seconds
INFO:lightwood-33203:[Learn phase 3/8] - Data splitting
INFO:lightwood-33203:Splitting the data into train/test
DEBUG:lightwood-33203: `split` runtime: 0.02 seconds
INFO:lightwood-33203:[Learn phase 4/8] - Preparing encoders
INFO:lightwood-33230:No torchvision detected, image helpers not supported.
INFO:lightwood-33228:No torchvision detected, image helpers not supported.
INFO:lightwood-33231:No torchvision detected, image helpers not supported.
INFO:lightwood-33229:No torchvision detected, image helpers not supported.
INFO:lightwood-33228:No torchvision/pillow detected, image encoder not supported
INFO:lightwood-33230:No torchvision/pillow detected, image encoder not supported
INFO:lightwood-33231:No torchvision/pillow detected, image encoder not supported
INFO:lightwood-33229:No torchvision/pillow detected, image encoder not supported
/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb-venv/lib/python3.8/site-packages/gluonts/json.py:101: UserWarning: Using `json`-module for json-handling. Consider installing one of `orjson`, `ujson` to speed up serialization and deserialization.
  warnings.warn(
/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb-venv/lib/python3.8/site-packages/gluonts/json.py:101: UserWarning: Using `json`-module for json-handling. Consider installing one of `orjson`, `ujson` to speed up serialization and deserialization.
  warnings.warn(
/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb-venv/lib/python3.8/site-packages/gluonts/json.py:101: UserWarning: Using `json`-module for json-handling. Consider installing one of `orjson`, `ujson` to speed up serialization and deserialization.
  warnings.warn(
/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb-venv/lib/python3.8/site-packages/gluonts/json.py:101: UserWarning: Using `json`-module for json-handling. Consider installing one of `orjson`, `ujson` to speed up serialization and deserialization.
  warnings.warn(
/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb-venv/lib/python3.8/site-packages/gluonts/model/deepar/__init__.py:18: FutureWarning: The module gluonts.model.deepar has been moved to gluonts.mx.model.deepar. In GluonTS v0.12 it will be no longer possible to use the old path. Try to use 'from gluonts.mx import DeepAREstimator'.
  warnings.warn(
/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb-venv/lib/python3.8/site-packages/gluonts/model/deepar/__init__.py:18: FutureWarning: The module gluonts.model.deepar has been moved to gluonts.mx.model.deepar. In GluonTS v0.12 it will be no longer possible to use the old path. Try to use 'from gluonts.mx import DeepAREstimator'.
  warnings.warn(
/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb-venv/lib/python3.8/site-packages/gluonts/model/deepar/__init__.py:18: FutureWarning: The module gluonts.model.deepar has been moved to gluonts.mx.model.deepar. In GluonTS v0.12 it will be no longer possible to use the old path. Try to use 'f/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb-venv/lib/python3.8/site-packages/gluonts/model/deepar/__init__.py:18: FutureWarning: The module gluonts.model.deepar has been moved to gluonts.mx.model.deepar. In GluonTS v0.12 it will be no longer possible to use the old path. Try to use 'from gluonts.mx import DeepAREstimator'.
  warnings.warn(
rom gluonts.mx import DeepAREstimator'.
  warnings.warn(
INFO:lightwood-33230:Encoding UNKNOWN categories as index 0
INFO:lightwood-33229:Encoding UNKNOWN categories as index 0
2022-11-27 23:39:22,230 - ERROR - Exception during query casting to 'postgres' dialect. Query: SELECT * FROM `models` WHERE (`predictors`.name = 'jira_predictor_model2') AND (project = 'mindsdb'). Error: 'Select' object has no attribute 'filter'
Traceback (most recent call last):
  File "/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb/mindsdb/api/mysql/mysql_proxy/utilities/sql.py", line 65, in query_df
    query_str = render.get_string(query_ast, with_failback=False)
  File "/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb-venv/lib/python3.8/site-packages/mindsdb_sql/render/sqlalchemy_render.py", line 556, in get_string
    stmt = self.prepare_select(ast_query)
  File "/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb-venv/lib/python3.8/site-packages/mindsdb_sql/render/sqlalchemy_render.py", line 409, in prepare_select
    query = query.filter(
AttributeError: 'Select' object has no attribute 'filter'
2022-11-27 23:39:22,321 - INFO - Traceback (most recent call last):
  File "/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb/mindsdb/api/mysql/mysql_proxy/utilities/sql.py", line 65, in query_df
    query_str = render.get_string(query_ast, with_failback=False)
  File "/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb-venv/lib/python3.8/site-packages/mindsdb_sql/render/sqlalchemy_render.py", line 556, in get_string
    stmt = self.prepare_select(ast_query)
  File "/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb-venv/lib/python3.8/site-packages/mindsdb_sql/render/sqlalchemy_render.py", line 409, in prepare_select
    query = query.filter(
AttributeError: 'Select' object has no attribute 'filter'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb/mindsdb/api/http/namespaces/sql.py", line 35, in post
    result = mysql_proxy.process_query(query)
  File "/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb/mindsdb/api/mysql/mysql_proxy/mysql_proxy.py", line 491, in process_query
    executor.query_execute(sql)
  File "/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb/mindsdb/api/mysql/mysql_proxy/executor/executor.py", line 109, in query_execute
    self.do_execute()
  File "/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb/mindsdb/api/mysql/mysql_proxy/executor/executor.py", line 189, in do_execute
    ret = self.command_executor.execute_command(self.query)
  File "/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb/mindsdb/api/mysql/mysql_proxy/executor/executor_commands.py", line 546, in execute_command
    query = SQLQuery(
  File "/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb/mindsdb/api/mysql/mysql_proxy/classes/sql_query.py", line 440, in __init__
    self.execute_query()
  File "/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb/mindsdb/api/mysql/mysql_proxy/classes/sql_query.py", line 656, in execute_query
    data = self.execute_step(step, steps_data)
  File "/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb/mindsdb/api/mysql/mysql_proxy/classes/sql_query.py", line 791, in execute_step
    data = self._fetch_dataframe_step(step)
  File "/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb/mindsdb/api/mysql/mysql_proxy/classes/sql_query.py", line 548, in _fetch_dataframe_step
    data, columns_info = dn.query(
  File "/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb/mindsdb/api/mysql/mysql_proxy/datahub/datanodes/project_datanode.py", line 89, in query
    data, columns_info = self.information_schema.query(new_query)
  File "/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb/mindsdb/api/mysql/mysql_proxy/datahub/datanodes/information_schema_datanode.py", line 411, in query
    data = query_df(dataframe, query, session=self.session)
  File "/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb/mindsdb/api/mysql/mysql_proxy/utilities/sql.py", line 70, in query_df
    query_str = render.get_string(query_ast, with_failback=True)
  File "/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb-venv/lib/python3.8/site-packages/mindsdb_sql/render/sqlalchemy_render.py", line 556, in get_string
    stmt = self.prepare_select(ast_query)
  File "/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb-venv/lib/python3.8/site-packages/mindsdb_sql/render/sqlalchemy_render.py", line 409, in prepare_select
    query = query.filter(
AttributeError: 'Select' object has no attribute 'filter'

INFO:lightwood-33227:No torchvision detected, image helpers not supported.
INFO:lightwood-33227:No torchvision/pillow detected, image encoder not supported
/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb-venv/lib/python3.8/site-packages/gluonts/json.py:101: UserWarning: Using `json`-module for json-handling. Consider installing one of `orjson`, `ujson` to speed up serialization and deserialization.
  warnings.warn(
/Users/bseetharaman/Desktop/Github_Repos_PRs/Mindsdb/mindsdb-venv/lib/python3.8/site-packages/gluonts/model/deepar/__init__.py:18: FutureWarning: The module gluonts.model.deepar has been moved to gluonts.mx.model.deepar. In GluonTS v0.12 it will be no longer possible to use the old path. Try to use 'from gluonts.mx import DeepAREstimator'.
  warnings.warn(
INFO:lightwood-33203:Done running for: Ticket_No
INFO:lightwood-33203:Done running for: Reporter
INFO:lightwood-33203:Done running for: Assignee
INFO:lightwood-33203:Done running for: Priority
INFO:lightwood-33203:Encoding UNKNOWN categories as index 0
INFO:lightwood-33203:Preparing a categorical autoencoder, this may take up to 85536.0 seconds.
INFO:lightwood-33203:Encoding UNKNOWN categories as index 0
INFO:lightwood-33203:Categorical autoencoder ready
DEBUG:lightwood-33203: `prepare` runtime: 18.6 seconds
INFO:lightwood-33203:[Learn phase 5/8] - Feature generation
INFO:lightwood-33203:Featurizing the data
DEBUG:lightwood-33203: `featurize` runtime: 0.0 seconds
INFO:lightwood-33203:[Learn phase 6/8] - Mixer training
INFO:lightwood-33203:Training the mixers
Found `num_iterations` in params. Will use it instead of argument

Segmentation fault: 11


Segmentation fault: 11


Segmentation fault: 11

libc++abi: terminating with uncaught exception of type std::__1::system_error: mutex lock failed: Invalid argument
^CStopping stream integrations...
Closing app...
```

### CODE REVIEW

File: mindsdb/integrations/handlers/jira_handler/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+The required arguments to establish a connection are,
+* `table_name`: Sample Jira table name.
+* `jira_url`: Jira  url
+* `user_id`: Jira User ID.
+* `api_key`: API key for the Jira API.
+* `jira_query`: Jira Search Query
+
+## Usage
+In order to make use of this handler and connect to an Access database in MindsDB, the following syntax can be used,
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/jira_handler/README.md:25]: Please change the `Access` database part

File: mindsdb/integrations/handlers/jira_handler/jira_handler.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+            password=self.connection_data['api_key'],
+            cloud=True)
+        results = jira_instance.jql(self.connection_data['jira_query'])
+        df = pd.json_normalize(results["issues"])
+        fields_name= ["key", "fields.summary","fields.status.name", "fields.reporter.name","fields.assignee.name","fields.priority.name"]
+        df = df[fields_name]
+        df = df.rename(columns={'key': 'Ticket_No', 'fields.summary': 'Summary','fields.reporter.name':'Reporter','fields.assignee.name':'Assignee','fields.priority.name':'Priority','fields.status.name':'Status'})
+
+
+        globals()[self.connection_data['table_name']] = df
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/jira_handler/jira_handler.py:74]: Do we need globals() here?

File: mindsdb/integrations/handlers/jira_handler/jira_handler.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+        jira_instance = Jira(
+            url=self.connection_data['jira_url'],
+            username=self.connection_data['user_id'],
+            password=self.connection_data['api_key'],
+            cloud=True)
+        results = jira_instance.jql(self.connection_data['jira_query'])
+        df = pd.json_normalize(results["issues"])
+        fields_name= ["key", "fields.summary","fields.status.name", "fields.reporter.name","fields.assignee.name","fields.priority.name"]
+        df = df[fields_name]
+        df = df.rename(columns={'key': 'Ticket_No', 'fields.summary': 'Summary','fields.reporter.name':'Reporter','fields.assignee.name':'Assignee','fields.priority.name':'Priority','fields.status.name':'Status'})
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/jira_handler/jira_handler.py:71]: Should we keep the original names? Or maybe add this to the docs so the user will know the names when querying the data.

File: mindsdb/integrations/handlers/jira_handler/jira_handler.py [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+        Returns:
+            HandlerResponse
+        """
+
+        response = Response(
+            RESPONSE_TYPE.TABLE,
+            data_frame=pd.DataFrame(
+                {
+                    'column_name': list(globals()[self.connection_data['table_name']].columns),
+                    'data_type': globals()[self.connection_data['table_name']].dtypes
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/jira_handler/jira_handler.py:199]: Do we need globals here too?

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]

File: mindsdb/integrations/handlers/jira_handler/README.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    "jira_query": "project = RELENG and status != Done"
+};
+~~~~
+
+Now, you can use this established connection to query your table as follows,
+~~~~sql
+SELECT * FROM jira_source.example_tbl
+~~~~
+
+At the moment, only `SELECT` queries are allowed to be executed through `duckdb`. This, however, has no restriction on running machine learning algorithms against your data in Airtable using `CREATE PREDICTOR` statements.
```
[the remark this answers is not in the document]
Comment (bala-ceg) [reply mindsdb/integrations/handlers/jira_handler/README.md]: Thanks for letting me know, changed it

File: mindsdb/integrations/handlers/jira_handler/README.md (hunk 1 above)
Comment (bala-ceg) [reply mindsdb/integrations/handlers/jira_handler/README.md]: Thanks for letting me know, I changed it

File: mindsdb/integrations/handlers/jira_handler/jira_handler.py (hunk 2 above)
Comment (bala-ceg) [reply mindsdb/integrations/handlers/jira_handler/jira_handler.py:72]: I am not access this data frame data without globals  in another functions

File: mindsdb/integrations/handlers/jira_handler/jira_handler.py (hunk 4 above)
Comment (bala-ceg) [reply mindsdb/integrations/handlers/jira_handler/jira_handler.py:196]: I am not access this data frame data without globals in another functions

File: mindsdb/integrations/handlers/jira_handler/jira_handler.py (hunk 3 above)
Comment (bala-ceg) [reply mindsdb/integrations/handlers/jira_handler/jira_handler.py]: Keeping the original names

**Comment by ZoranPandovski** [comment]
@bala-ceg As discussed on slack the next step is to remove globals() and setup the config_data in the constructor

**Comment by bala-ceg** [comment]
@ZoranPandovski  made the changes, can you please review the PR based on the latest commit?

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]
Can you make sure that this PR is up to date with the latest staging? It looks like the tests are failing but they were resolved on the latest staging

**Comment by ZoranPandovski** [comment]
Awesome, please also try if the `query` works now?

**Comment by bala-ceg** [comment]
yes @ZoranPandovski the query works after updating the branch

Screenshot 2022-12-19 at 7 57 09 PM
Screenshot 2022-12-19 at 8 00 39 PM

**Pull request merged**